app/assets/src/components/PipelineSampleReport.jsx

```diff
@@ -1205,6 +1211,57 @@ class PipelineSampleReport extends React.Component {
     );
   }
 
+  // Fill in desired URL parameters so user's can copy and paste URLs.
+  // Ex: Add ?pipeline_version=1.7&background_id=4 to /samples/545
+  // This way links can still be to '/samples/545' in the rest of the app
+  // but the URL will be filled in without triggering another page reload.
+  //
+  // Order of precedence for background_id is:
+  // (1) URL parameter specified
+  // (2) saved background name in frontend cookie
+  // (3) the default background
+  fillUrlParams() {
+    // Skip if report is not present or a background ID and pipeline version
+    // are explicitly specified in the URL.
+    if (
+      !this.props.reportPageParams ||
+      (this.fetchParams("pipeline_version") &&
+        this.fetchParams("background_id"))
+    ) {
+      return;
+    }
+
+    // Setup
+    let params = new URLSearchParams(window.href);
+    const stringer = require("querystring");
+
+    // If a background name is set in Cookies, get its ID from allBackgrounds.
+    // This is necessary because soon we may show different backgrounds IDs
+    // as the same name to the users.
+    if (!this.fetchParams("background_id") && Cookies.get("background_name")) {
+      this.getBackgroundIdByName(Cookies.get("background_name"));
+    }
+
+    // Set pipeline_version and background_id from reportPageParams.
+    params["pipeline_version"] = this.props.reportPageParams.pipeline_version;
+    params["background_id"] = this.props.reportPageParams.background_id;
+    // Modify the URL in place without triggering a page reload.
+    history.replaceState(null, null, `?${stringer.stringify(params)}`);
+  }
+
+  fetchParams(param) {
+    let url = new URL(window.location);
+    return url.searchParams.get(param);
+  }
+
+  // Select the background with the matching name.
+  getBackgroundIdByName(name) {
+    let match = this.all_backgrounds.filter(b => b["name"] === name);
+    if (match && match[0] && match[0]["id"]) {
+      this.props.reportPageParams.background_id = match[0]["id"];
```

Comment:
you can probably just return match[0]["id"] 
and do 
const background_id = getBackgroundIdByName(name) ; 
if (background_id) this.props.reportPageParams.background_id = background_id 

so getBackgroundIdByName(name) might be reusable in the future.